app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,42 @@
+class CategoriesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_category, only: %i[edit update destroy]
+  before_action :ensure_user_owns_category, only: %i[edit update destroy]
+
+  def index
+    @categories = current_user.categories
+    # @total_purchases = current_user.purchases.pluck(:amount).sum
+    # @total_purchases = current_user.purchases.joins(:categories).pluck(:amount).sum
+    @total_purchases = current_user.purchases.includes(:categories).pluck(:amount).sum
```

Comment:
- [ ] Kindly remove commented lines of code for a cleaner codebase

Reply:
Done @Munsa1 Thank you.